app/src/actions/rescindAndUnstake.ts

```diff
@@ -109,11 +106,43 @@ export const rescindAndUnstake = async (
     unbondingSeed,
     amount
   );
-
   allTxs.push({
     tx: new Transaction().add(...unbondIxs),
     signers: []
   });
 
+  // If there is still remaining JET and they have previously cast votes,
+  // Re-cast those votes if proposals are still active
+  if (voteRecords && tokenOwnerRecord && !unstakeAll) {
+    for (const i in voteRecords) {
+      let proposal = proposals[voteRecords[i].account.proposal.toString()];
+      if (!proposal || proposal.account.hasVoteTimeEnded(governance.account)) {
+        continue;
+      }
+      if (voteRecords[i].account.vote) {
+        const recastIxs: TransactionInstruction[] = [];
+        await withCastVote(
+          recastIxs,
+          programId,
+          programVersion,
+          stakePool.stakePool.governanceRealm,
+          proposal.account.governance,
+          proposal.pubkey,
+          proposal.account.tokenOwnerRecord,
+          tokenOwnerRecord.pubkey,
+          walletPubkey,
+          proposal.account.governingTokenMint,
+          voteRecords[i].account.vote!,
```

Comment:
You can pull voteRecords[i].account.vote into it's own variable next to const proposal and check if it's undefined
```suggestion
          voteRecords[i].account.vote,
```